### ISSUE TITLE
"Publication has been modified by another user. Please reload the page." keep appearing even if nobody else is inside the document.

### ISSUE THREAD

**Issue by mietcls** [ugent-library/biblio-backoffice#1012]
# Bug

"Publication has been modified by another user. Please reload the page." keep appearing even if nobody else is inside the document.

![Image](https://user-images.githubusercontent.com/1572885/216614710-38ce5139-b159-4168-9451-2271cb84a29b.png)

## Steps to reproduce

It's hard to find a pattern, but it even happens during demo's when they're creating records from scratch and nobody else is editing.

The patterns up until now:
- When deleting a department – and none of the departments remain
- Often when there's a record starting from scratch
- When deleting a link

**Comment by nicolasfranck** [comment]
Things to do in order to further investigate this:

* Log when there is a conflict error (there is none now, it just renders a response for the end user). That way we track the flow.
* Log request parameters. Maybe the header `If-Match` is not sent, and it is a browser issue?

**Comment by mietcls** [comment]
Happened today with one of our reviewers:
![Image](https://user-images.githubusercontent.com/1572885/217575549-02f2e7a2-2d8e-4d11-bfea-91c787d60b55.png)

- wanted to remove the only department in the list
- when they checked the last edit, it was edited by themselves, not by anyone else
- **the record was open in two tabs** – but only one had edits
- the record wasn't open longer than usual

Would having the record open in two tabs, but only one is being edited, still trigger that notification?

**Comment by nicolasfranck** [comment]
Well, if you have a modal open in both tabs, belonging to the same record, then you will get this error.

e.g. you open the "edit details" modal but do nothing in one tab, and in another tab you try to remove a link. What happens is that in the second tab new version is created, while tab one is still working on a previous version. If you then would "save" in that tab, you will get the error. Let's not forget that all of the data is located in one and the same record, so reasons like "I was not editing that does not apply".

In short: if you have opened a modal, do not try to edit something of that same record using another way.

**Comment by nicolasfranck** [comment]
@mietcls does that make sense?

**Comment by mietcls** [comment]
@nicolasfranck Yes, it does, thank you for explaining! The modal wasn't open, just the record. It was not in edit mode. So it should not have triggered it, correct?

**Comment by nicolasfranck** [comment]
Another, similar use case: 

* open tab "Description" and tab "Files" each in a separate browser tab.
* Click "Edit" in the "Description" tab, and then click "Save"
* Go to the second browser tab, and try to upload a file: you'll get an error dialog saying "Publication has been modified by another user. Please reload the page."

As said above, it would be handy to add a warning dialog (or a less intrusive flash warning) that warns you about a possible future version conflict:

* as soon as a browser tab is focused, a call is done to the server with the current version contents. The server responds back if that version is still valid.
* if the version is outdated, a non intrusive flash message is shown to the user. "your page is outdated, content has been updated". I can imagine of course that such a message is not welcome if it appears on your file upload page, the upload takes a long time, and then such message appears, possibly followed by the previous modal "publication has been modified.." when the upload ends. In that case two messages are sent.
